internal/services/users/user_resource.go

```diff
@@ -514,6 +547,13 @@ func userResourceUpdate(ctx context.Context, d *schema.ResourceData, meta interf
 		properties.OnPremisesImmutableId = utils.String(d.Get("onpremises_immutable_id").(string))
 	}
 
+	if d.HasChange("cost_center") || d.HasChange("division") {
+		properties.EmployeeOrgData = &msgraph.EmployeeOrgData{
+			CostCenter: utils.String(d.Get("cost_center").(string)),
+			Division:   utils.String(d.Get("division").(string)),
+		}
+	}
+
```

Comment:
As above, when updating we should also set these every time. _Generally_, we only need to use `d.HasChange()` for Optional+Computed fields.

Reply:
done, good point!